features/aa/aa_features/sponsorship.feature

```diff
@@ -66,13 +65,16 @@ Feature:
     And I should see "Second Orphan"
     And I fill in Sponsorship Start Date for "First Orphan" with "2014-01-31"
     When I click the "Sponsor this orphan" link for orphan "First Orphan"
+    Then I should be on the Link to Orphan page for sponsor "First Sponsor"
+    And I should see "Return to Sponsor page"
+    And I should see "Sponsorship link was successfully created for First Orphan"
+    And I fill in Sponsorship Start Date for "Second Orphan" with "2014-01-31"
+    When I click the "Sponsor this orphan" link for orphan "Second Orphan"
+    Then I should be on the Link to Orphan page for sponsor "First Sponsor"
```

Comment:
About your first concern, yes, I should add test about the count. 
About your second concern. No, it won't redirect, because when the sponsor are created they are defined as `requested_orphan_count: 5`. See this: https://github.com/gmunumel/osra/blob/OSRA387/features/aa/step_definitions/aa_steps/sponsorship_steps.rb

Can I try to change this value and maybe broke some test, or add 3 more orphans.